pkg/services/ngalert/notifier/channels/webex.go

```diff
@@ -0,0 +1,175 @@
+package channels
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/alertmanager/types"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	ngmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/notifications"
+)
+
+const webexAPIEndpoint = "https://webexapis.com/v1/messages"
+
+// WebexNotifier is responsible for sending alert notifications as webex messages.
+type WebexNotifier struct {
+	*Base
+	ns       notifications.WebhookSender
+	log      log.Logger
+	images   ImageStore
+	tmpl     *template.Template
+	orgID    int64
+	settings *webexSettings
+}
+
+// PLEASE do not touch these settings without taking a look at what we support as part of
+// https://github.com/prometheus/alertmanager/blob/main/notify/webex/webex.go
+// Currently, the Alerting team is unifying channels and receivers - any discrepancy is detrimental to that.
+type webexSettings struct {
+	Message    string `json:"message,omitempty" yaml:"message,omitempty"`
+	RoomID     string `json:"room_id,omitempty" yaml:"room_id,omitempty"`
+	WebhookURL string `json:"webhook_url,omitempty" yaml:"webhook_url,omitempty"`
+	Token      string `json:"bot_token" yaml:"bot_token"`
```

Comment:
This is my most recent discovery - looking at the Webex documentation it seems that the recommended approach is to use a "Bot", the "Bot" comes with a Token that you can use to interact with the Webex rooms. 

As such, the required configuration appears to be: A bot token and which room you want to send alerts too (1 at a time).

I couldn't figure out a way to include the token/room as part of the URL as I previously thought, which means that we probably don't need to encrypt the URL and just the token?